[stable/jenkins] Adding pod env vars, image pull secrets and fixes to some values

ecejjar opened the pull request:
We've cooked this helm chart for our own Jenkins-based CI environment. The main improvements are:

- additional environment variables can be defined for the slaves
- image pull secrets for private Docker registries can be used by a slave
- one minor enhancement to be able to use the BuildInfo proxy in the Artifactory plugin
- Retention, Connect and Read timeouts can be specified for master-slave interaction
- fixes in a couple of field values that were preventing us from using our own Docker image for the slaves

In connection with PR#1389, this PR only allows image pull secrets in slaves since the master is not supposed to do any work at all.

Commit:
Added por env vars, image pull secrets and some field values

Added the ability to define pod environment variables in helm chart values (Values->Agent->envVars)
Added the ability to define image pull secrets (Values->Agent->imagePullSecrets)
Added the ability to set retention time-out and connect/read time-outs
Upgraded plugin version (1.11->1.12)
Fixed pod template namespace and default provider template values

Commit:
Added container port for Artifactory's BuildInfo proxy

When using the Artifactory plug-in, a BuildInfo proxy is necessary to build Docker images.
This enhancement allows opening the proxy port in the Jenkins master.

Commit:
Update values.yaml

Commit:
Update README.md

Commit:
Merge pull request #1 from ecejjar/jenkins-chart-improvements

Added pod env vars, image pull secrets and some field values

Commit:
Added configuration of idleMinutes and slaveConnectTimeout

SlaveConnectTimeout should be default set to 100 according to on-screen Jenkins GUI help, but it is set to 0. If set to 0 no slaves pods are ever started in our k8s deployments.
Slave pods are reaped after ~10min even if they're not done with their jobs yet. Increasing IdleTimeout reduces pod reaping for long-running jobs.

Commit:
Merge of upstream master onto local master branch

Commit:
Bring back elements from previous versions

Some configuration elements removed in a recent commit render our Jenkins master unable to launch slaves for jobs scheduled from an init script.
When those elements are brought back into the config.yaml the Jenkins master handles those jobs seamlessly.

tommy-dk commented on stable/jenkins/values.yaml:
```diff
@@ -41,6 +41,8 @@ Master:
+# BuildInfoProxyPort: 9000
 # List of plugins to be install during Jenkins master start
   InstallPlugins:
       - kubernetes:0.11
```
Should this not be 0.12? As that is the version defined in config.yaml?

ecejjar commented:
That's correct, I missed updating that value.

I'm submitting a new commit that fixes it.

Thanks!
/Javi

On sáb, 2017-09-16 at 03:15 -0700, Tommy Heyde Olsen wrote:
> @tommy-dk commented on this pull request.
> In stable/jenkins/values.yaml:
> > @@ -41,6 +41,8 @@ Master:
>  #   -Dcom.sun.management.jmxremote.authenticate=false
>  #   -Dcom.sun.management.jmxremote.ssl=false
>  # JMXPort: 4000
> +# Optionally configure a BuildInfoProxy port for artifactory plugin
> +# BuildInfoProxyPort: 9000
>  # List of plugins to be install during Jenkins master start
>    InstallPlugins:
>        - kubernetes:0.11
> Should this not be 0.12? As that is the version defined in
> config.yaml?
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

Commit:
Updated k8s plugin version

The kubernetes plugin version should match that in templates/config.yaml.
Thanks to Tommy Heyde for pointing this out.

viglesiasce commented:
/ok-to-test

viglesiasce commented on stable/jenkins/templates/jenkins-master-deployment.yaml:
```diff
@@ -87,6 +87,10 @@ spec:
             - containerPort: {{ .Values.Master.JMXPort }}
               name: jmx
             {{- end }}
+            {{- if .Values.Master.BuildInfoProxyPort }}
```
Can we make this generic? Something like:

```
Master:
  # AddtionalPorts:
  # - Name: arm-proxy
  #   ContainerPort:   9099
```

I'm trying to avoid having to do this for each plugin that needs something like it.

ecejjar replied:
Sounds good. Will try to include it in next commit.

viglesiasce commented on stable/jenkins/values.yaml:
```diff
@@ -41,9 +41,11 @@ Master:
 # List of plugins to be install during Jenkins master start
   InstallPlugins:
-      - kubernetes:0.11
+      - kubernetes:0.12
```
Can we move the update of the plugin to a separate PR? 

Looks like this also causes changes to the config.xml that makes it hard to decipher what is needed. Additionally it was not mentioned in the description so not sure if this is required/intended for the changes listed there.

viglesiasce commented on stable/jenkins/templates/config.yaml:
```diff
@@ -78,9 +85,34 @@ data:
+{{- end }}
+              </envVars>
+              <ports/>
+              <livenessProbe>
```
Here are examples of new config params that arent documented or maybe even needed (for example can we just use the defaults).

ecejjar replied:
Agree. Will include in next commit.

viglesiasce commented on stable/jenkins/templates/config.yaml:
```diff
@@ -43,6 +45,11 @@ data:
                   {{- $key }}={{ $value }}
                   {{- $_ := set $local "first" false }}
                 {{- end }}</nodeSelector>
+              <nodeUsageMode>NORMAL</nodeUsageMode>
```
More config params that were not here before.

ecejjar replied:
Right. Will include in next commit.

viglesiasce reviewed (changes requested):
Looks great! Just would like to make the additional port more flexible and tease out the config changes and kubernetes plugin update.

viglesiasce commented:
Marking stale.

Commit:
Merge branch 'master' into master

ecejjar commented:
Fixed merge conflicts. The changes break deployments using "ImagePullSecret" in values.yaml; just replacing the "ImagePullSecret" value by a list "ImagePullSecrets" will make the trick.

ecejjar commented:
CI failed due to outdated version in chart, fixed in new commit.

Commit:
Bumped up chart version

Merging new commits after rebasing from master causes CI failure. Bumping up the chart version should fix that.

viglesiasce commented:
@ecejjar I'm still seeing changes to the config that don't line up with the changes to the chart that you are introducing. Can you take a look?

viglesiasce commented:
Closing as stale.
